File: model/Card.cs
using Model.Common;

namespace Model
{
    public class Card : PocketObject
    {
        private readonly CardType _type;

        public Card(int id, CardType type) : base(id)
        {
            _type = type;
        }

        public CardType Type
        {
            get { return _type; }
        }

        public override string ToString()
        {
            return $"Card{{type={_type}}}";
        }
    }
}

File: model/Card.js
import PocketObject from './common/PocketObject.js';


class Card extends PocketObject {
    constructor(id, type) {
        super(id);
        this._type = type;
    }

    getType() {
        return this._type;
    }

    toString() {
        return `Card{type=${this._type}}`;
    }
}

export default Card